[chonguschonguschongus] iP

chonguschonguschongus opened the pull request:
# DUKEYDUKEYDUKEY
> "Time flies like an arrow, fruit fly like banana" - Unknown

Is this you? 
- [x] I want to be _free_ of deadlines
- [x] I want to _enjoy_ my life
- [x] I want _control_ over my time

Then, Dukey is here to help manage your time so well you will never forget another deadline. It can help log your:
- to-dos
- deadlines
- events
- so much more!

Want Dukey **now**? Simply follow these steps:
1. Download Dukey 💻 
2. Launch Dukey 🚀 
3. Start Duking!! 😋 

[Contribute](https://i.imgur.com/hIsG3Mo.jpg) to the project! 

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
Level 0 changes1

Commit:
Level 1 changes

Commit:
added list function and null entry reply

Commit:
added mark/unmark function and Task class

Commit:
added try catch blocks for IndexOutOfBoundException and created new classes that inherit task

Commit:
added abstraction for Task child classes and ability to add different types of tasks to the list

Commit:
added Automated Text UI Testing

Commit:
added error handling for incorrect inputs like missing task descriptions/deadlines, updated test files

Commit:
added Delete function and updated test files

chonguschonguschongus committed:
Add FileHandler class and ability to write tasklist to a file as well as more exception catching

chonguschonguschongus committed:
Change some of Dukey's replies

chonguschonguschongus committed:
Deconflict

chonguschonguschongus committed:
Add TimeFormatter to handle formatting of inputted dates

chonguschonguschongus committed:
Merge branch 'branch-Level-8' Add TimeFormatter

chonguschonguschongus committed:
Extract classes out for more OOP

chonguschonguschongus committed:
Merge branch 'branch-A-MoreOOP'

chonguschonguschongus committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

chonguschonguschongus committed:
Add Gradle support

chonguschonguschongus committed:
Add JUnit tests for toString() and parse() methods

chonguschonguschongus committed:
Revert "Add JUnit tests for toString() and parse() methods"

This reverts commit 3e48ee6f10d850bb08eafe0775cf2e30e80387f6.

chonguschonguschongus committed:
Revert "Revert "Add JUnit tests for toString() and parse() methods""

This reverts commit 767d0ceac35b6fe42030d6c45edaaac490e8d084.

chonguschonguschongus committed:
Revert "Add JUnit tests for toString() and parse() methods"

This reverts commit 3e48ee6f10d850bb08eafe0775cf2e30e80387f6.

chonguschonguschongus committed:
Merge branch 'branch-A-JUnit'

chonguschonguschongus committed:
Add JUnit tests for toString() and parse() methods

chonguschonguschongus committed:
Reset branch

chonguschonguschongus committed:
Merge branch 'branch-A-JUnit'

chonguschonguschongus committed:
Add test files to src

chonguschonguschongus committed:
Merge branch 'master' of github.com:chonguschonguschongus/ip

chonguschonguschongus committed:
Refactor some files

chonguschonguschongus committed:
Add Jar file

chonguschonguschongus committed:
Add JavaDoc

chonguschonguschongus committed:
Fix according to Coding Standard

chonguschonguschongus committed:
Add Find function

chonguschonguschongus committed:
Merge branch 'branch-Level-9'

chonguschonguschongus committed:
Fix merge conflict

chonguschonguschongus committed:
Fix merge conflicts

coderhuang559 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,107 @@
+import java.util.Scanner;
+
+public class Ui {
+    private static String logo =
```
Wow! Very creative logo, chonguschonguschongus! 😁👌👍

chonguschonguschongus replied:
hi coder! do consider supporting the project if you really like it :) [buy me a haidilao](https://i.imgur.com/hIsG3Mo.jpg)

coderhuang559 commented on src/main/java/command/Command.java:
```diff
@@ -0,0 +1,43 @@
+ * The base class for all command implementations in the Duke application.
+ * Represents a command that can be executed on the task list.
+ */
+public abstract class Command {
```
Well-implemented & clean code. Keep it up!

> The best preparation for good work tomorrow is to do good work today. -Elbert Hubbard

coderhuang559 reviewed (commented):
LGTM. Great job!

chonguschonguschongus committed:
Added basic JavaFX tutorial features

Leb14 commented on src/main/java/duke/Ui.java:
```diff
@@ -1,7 +1,5 @@
-
 import java.util.Scanner;
 
 public class Ui {
```
The logo looks really impressive, and it's clear you put in a lot of effort. Great job, and keep up the good work! :fire::fire::fire:

Leb14 commented on src/main/java/duke/TaskList.java:
```diff
@@ -5,52 +5,104 @@
+     */
     public TaskList findTask(String query) {
         TaskList queryList = new TaskList();
         for (Task t : this.todoList) {
```
I believe that eliminating the 'continue' statement should not impact the code's logic. But overall the logic of finding a Task is pretty good! :rocket::rocket:

Leb14 reviewed (commented):
In general, the adherence to OOP principles and coding standards in the codebase is commendable. Great job, chonguschonguschongus! :green_heart: Keep up the excellent work!

chonguschonguschongus commented:
> In general, the adherence to OOP principles and coding standards in the codebase is commendable. Great job, chonguschonguschongus! 💚 Keep up the excellent work!

If you really like the project please [contribute](https://i.imgur.com/hIsG3Mo.jpg)

chonguschonguschongus committed:
Added Checkstyle XML files

chonguschonguschongus committed:
Merge branch 'branch-A-Checkstyle'

nicleongyj commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,107 @@
+                    " |[][][][][][][][][]|    |  \\..  |\n" +
+                    " \\------------------/    | ( # ) |\n" +
+                    "                         |  '''  |\n" +
+                    "                         \\_______/";
```
Very nice and creative logo! Good job!

chonguschonguschongus replied:
Thanks! 💯

nicleongyj commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,107 @@
+     * @param taskMarked The task that was marked or unmarked.
+     * @param isMark     Whether the task was marked (true) or unmarked (false).
+     */
+    public void showMarkTaskMessage(String taskMarked, Boolean isMark) {
```
Maybe for isMark should be renamed to isMarked?

chonguschonguschongus replied:
Good catch! Thanks!

nicleongyj commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,107 @@
+     * @param isMark     Whether the task was marked (true) or unmarked (false).
+     */
+    public void showMarkTaskMessage(String taskMarked, Boolean isMark) {
+        if (isMark = true) {
```
`if (isMarked) { body }` will be better?

chonguschonguschongus replied:
Didn't realise its redundant. Thanks for the heads up 😄

nicleongyj reviewed (commented):
Overall good job in your code styling! Just take note you can simplify some of your conditional if statements to just if (isDone). Keep up the awesome work!

chonguschonguschongus committed:
Integrate chatbot logic into basic GUI

chonguschonguschongus committed:
Merge branch 'branch-Level-10'

chonguschonguschongus committed:
Rewrite Parser class for better code quality

Current Parse funtion in Parser class is hundreds of lines
long and is difficult to read.

Extracting possible lower-level functions can reduce the
length of the method and improve readability.

Let's write methods for constructing different command types so
.parse() does not need to do all the parsing within itself
but call other methods to do so, reducing code length.

chonguschonguschongus committed:
Rewrote Parser class to avoid long methods and overnesting

chonguschonguschongus committed:
Add read from file function

chonguschonguschongus committed:
Add read from file function

chonguschonguschongus committed:
Add ability to update task information.

Chatbot has no way of updating task information should
there be a need to change task title, deadlines etc.

It will be inconveneint for users who key in wrong information.

Allowing users to change the task fields can result in more
convenience as well as better user experience.

Let's add a new Command that will call methods to edit Task
information according to which field the user inputs.

chonguschonguschongus committed:
Add ability to update task title, deadlines and timings

chonguschonguschongus committed:
Merge branch 'branch-A-Assertions'

chonguschonguschongus committed:
Merge branch 'branch-C-Update'

chonguschonguschongus committed:
Add more assertions

chonguschonguschongus committed:
Add more assertions

chonguschonguschongus committed:
Merge pull request #2 from chonguschonguschongus/branch-A-MoreAssertions

Add more assertions

chonguschonguschongus committed:
Clean up code

chonguschonguschongus committed:
Add JavaDoc to command

chonguschonguschongus committed:
Amend coding quality

chonguschonguschongus committed:
Resolve conflicts

chonguschonguschongus committed:
Add more personality

chonguschonguschongus committed:
Merge branch 'branch-A-MorePersonality'

chonguschonguschongus committed:
Improve GUI

chonguschonguschongus committed:
Merge branch 'branch-A-BetterGUI'

chonguschonguschongus committed:
Update README.md

Update user guide

chonguschonguschongus committed:
Update README.md

chonguschonguschongus committed:
Update README.md

chonguschonguschongus committed:
Update README.md

chonguschonguschongus committed:
Add more personality

chonguschonguschongus committed:
Update README.md

chonguschonguschongus committed:
Update README.md

chonguschonguschongus committed:
Add UI screenshot

chonguschonguschongus committed:
Merge branch 'master' of github.com:chonguschonguschongus/ip

chonguschonguschongus committed:
Delete unused files

chonguschonguschongus committed:
Update README.md

chonguschonguschongus committed:
Add title to window and update Ui pic

chonguschonguschongus committed:
Merge branch 'master' of github.com:chonguschonguschongus/ip

chonguschonguschongus committed:
Fix testfiles
